src/daft-micropartition/src/partitioning.rs

```diff
@@ -26,7 +27,9 @@ impl Partition for MicroPartition {
 // An in memory partition set
 #[derive(Debug, Default, Clone)]
 pub struct MicroPartitionSet {
-    pub partitions: DashMap<PartitionId, MicroPartitionRef>,
+    // We need ordering of partitions for the output
+    // TODO: Look into using ConcurrentMap if performance is an issue
```

Comment:
Since we're replacing the Python LocalPartitionSet with this, we need to maintain ordering like Python's dictionaries. I don't think this is a perf bottleneck but there are options if it turns out to be.